bindings/java/src/org/sleuthkit/datamodel/SleuthkitJNI.java

```diff
@@ -101,7 +103,8 @@ private SleuthkitJNI() {
 		 * @param fsHandle   The file system handle in which the file lives.
 		 */
 		private static void addFileHandle(long fileHandle, long fsHandle) {
-			synchronized (cacheLock) {
+			cacheLock.writeLock().lock();
```

Comment:
This can probably be a read lock. I think we need the write block only when we are looking to remove something.

Reply:
Our caches are just maps so I was trying to protect every place that accesses them. We used to have synchronized blocks any time the map was changed or accessed. Although I'm not actually sure the locks will ensure the map is properly visible on different threads. If we changed the maps to concurrent hash maps then we could probably have only the case closing section get the write lock.

Reply:
Oh yea.  Worth a try.